packages/react-reconciler/src/ReactFiberLane.new.js

```diff
@@ -742,6 +747,53 @@ export function getBumpedLaneForHydration(
   return lane;
 }
 
+export function addFiberToLanesMap(
+  root: FiberRoot,
+  fiber: Fiber,
+  lanes: Lanes | Lane,
+) {
+  if (enableUpdaterTracking) {
+    if (isDevToolsPresent) {
+      const pendingUpdatersLaneMap = root.pendingUpdatersLaneMap;
+      while (lanes > 0) {
+        const index = laneToIndex(lanes);
+        const lane = 1 << index;
+
+        const updaters = pendingUpdatersLaneMap[index];
+        updaters.add(fiber);
+
+        lanes &= ~lane;
+      }
+    }
+  }
+}
+
+export function movePendingFibersToMemoized(root: FiberRoot, lanes: Lanes) {
+  if (enableUpdaterTracking) {
+    if (isDevToolsPresent) {
+      const pendingUpdatersLaneMap = root.pendingUpdatersLaneMap;
+      const memoizedUpdaters = root.memoizedUpdaters;
+      while (lanes > 0) {
+        const index = laneToIndex(lanes);
+        const lane = 1 << index;
+
+        const updaters = pendingUpdatersLaneMap[index];
+        if (updaters.size > 0) {
+          updaters.forEach(fiber => {
+            const alternate = fiber.alternate;
+            if (alternate === null || !memoizedUpdaters.has(alternate)) {
+              memoizedUpdaters.add(fiber);
+            }
+          });
+          updaters.clear();
+        }
+
+        lanes &= ~lane;
+      }
+    }
+  }
+}
+
```

Comment:
It seemed best to put these functions in `ReactFiberLane` so they had access to `laneToIndex` to make lanes->indices mapping easier.